src/topology/instances/add_circle.lean

```diff
@@ -286,3 +286,115 @@ local attribute [instance] fact_zero_lt_one
 /-- The unit circle `ℝ ⧸ ℤ`. -/
 @[derive [compact_space, normal_space, second_countable_topology]]
 abbreviation unit_add_circle := add_circle (1 : ℝ)
+
+
+section identify_Icc_ends
+/-! This section proves that the natural map from `[0, T] ⊂ ℝ` to `add_circle T` gives an
+identification of `add_circle T`, as a topological space, with the quotient of `[0, T]` by the
+equivalence relation identifying `0` with `T`. -/
+
+variables (T : ℝ) [hT : fact (0 < T)]
+include hT
+local notation `𝕋` := add_circle T
+
+private lemma Icc_to_circle_kernel {x y : ℝ} (hx : x ∈ Icc 0 T) (hy : y ∈ Icc 0 T) :
```

Comment:
What are your thoughts on generalizing this to 
```suggestion
private lemma Icc_to_circle_kernel {x y : ℝ} (hx : x ∈ Icc a (a + T)) (hy : y ∈ Icc a (a + T)) :
```

Reply:
This would be easy enough to do, I suppose.